builtin/providers/aws/resource_aws_lambda_permission.go

```diff
@@ -182,7 +182,14 @@ func resourceAwsLambdaPermissionRead(d *schema.ResourceData, meta interface{}) e
 		statement, err = findLambdaPolicyStatementById(&policy, d.Id())
 		return err
 	})
+
 	if err != nil {
+		if _, isTimeout := err.(resource.ErrTimeout); isTimeout {
```

Comment:
I don't think we should be just checking for "timeout" error here, I think we'll need to be more specific, just in case we'd be timing out for different reasons. We only want to wipe the resource if it _actually_ doesn't exist.